casepro/pods/base.py

```diff
@@ -0,0 +1,46 @@
+from confmodel import fields, Config as ConfmodelConfig
+from django.apps import AppConfig
+from django.conf.urls import url
+from django.http import HttpResponse
+
+
+class PodConfig(ConfmodelConfig):
+    '''
+    This is the config that all pods should use as the base for their own
+    config.
+    '''
+    index = fields.ConfigInt(
+        "A unique identifier for the specific instance of this pod."
+        "Automatically determined and set in the pod registry.",
+        required=True)
+
+
+class Pod(object):
+    '''
+    The base class for all pod plugins.
+    '''
+    config_cls = PodConfig
+    name = 'Default pod name'
+    controller = None
+    directive = None
+
+    def __init__(self, config):
+        self.config = self.config_cls(config)
+
+    @property
+    def url_patterns(self):
+        return (
+            url(r'read/%d/$' % self.config.index, self.request_callback),
```

Comment:
I thinking of having a `urls.py` that looks like:
```python
url_patterns = [
   url(r'read/%(pk)s/$', views.read_pod_data),
   url(r'action/(%pk)s/$`, views.perform_pod_action),
]
```
and then a `views.py` that looks like:
```python
def read_pod_data(request, pk):
    pod = get_pod(pk)
    data = pod.read_data()
    return HttpResponse(json.dumps(data))

def perform_pod_action(request, pk):
    pod = get_pod(pk)
    params = json.loads(request.body)
    result = pod.perform_action(params)
    return HttpResponse(json.dumps(data))
```